packages/core/src/libraries/hook/index.ts

```diff
@@ -88,51 +138,54 @@ export const createHookLibrary = (queries: Queries) => {
       userIp,
       user: user && pick(user, ...userInfoSelectFields),
       application: application && pick(application, 'id', 'type', 'name', 'description'),
-    } satisfies Omit<HookEventPayload, 'hookId'>;
+    } satisfies BetterOmit<InteractionHookEventPayload, 'hookId'>;
 
-    await Promise.all(
-      rows.map(async ({ id, config, signingKey }) => {
-        consoleLog.info(`\tTriggering hook ${id} due to ${hookEvent} event`);
-        const json: HookEventPayload = { hookId: id, ...payload };
-        const logEntry = new LogEntry(`TriggerHook.${hookEvent}`);
+    await sendWebhooks(
+      hooks.map((hook) => ({ hook, payload })),
+      consoleLog
+    );
+  };
 
-        logEntry.append({ hookId: id, hookRequest: { body: json } });
+  /**
+   * Trigger data hooks with the given data mutation context. All context objects will be used to trigger hooks.
+   */
+  const triggerDataHooks = async (consoleLog: ConsoleLog, hooksManager: DataHookContextManager) => {
+    if (hooksManager.contextArray.length === 0) {
+      return;
+    }
 
-        // Trigger web hook and log response
-        await sendWebhookRequest({
-          hookConfig: config,
-          payload: json,
-          signingKey,
-        })
-          .then(async (response) => {
-            logEntry.append({
-              response: await parseResponse(response),
-            });
-          })
-          .catch(async (error) => {
-            logEntry.append({
-              result: LogResult.Error,
-              response: conditional(
-                error instanceof HTTPError && (await parseResponse(error.response))
-              ),
-              error: conditional(error instanceof Error && String(error)),
-            });
-          });
+    const found = await findAllHooks();
 
-        consoleLog.info(
-          `\tHook ${id} ${logEntry.payload.result === LogResult.Success ? 'succeeded' : 'failed'}`
+    // Filter hooks that match each events
+    const webhooks = hooksManager.contextArray
+      .map(({ event, data }) => {
+        const hooks = found.filter(
+          ({ event: hookEvent, events, enabled }) =>
+            enabled && (events.length > 0 ? events.includes(event) : event === hookEvent)
         );
 
-        await insertLog({
-          id: generateStandardId(),
-          key: logEntry.key,
-          payload: logEntry.payload,
-        });
+        if (hooks.length === 0) {
+          return;
+        }
+
+        const payload = {
+          event,
+          createdAt: new Date().toISOString(),
+          ...hooksManager.metadata,
+          ...data,
+        } satisfies BetterOmit<DataHookEventPayload, 'hookId'>;
+
+        return { hooks, payload };
       })
+      .filter(Boolean);
+
+    await sendWebhooks(
+      webhooks.flatMap(({ hooks, payload }) => hooks.map((hook) => ({ hook, payload }))),
```

Comment:
no need to use `.flatMap` here if we have `const webhooks = hooksManager.contextArray.flatMap`?